lib/util/viewer.js

```diff
@@ -47,20 +49,48 @@ export function routeIsValid (route, routeId) {
   return true
 }
 
+/**
+ * Run heuristic on pattern description to extract headsign from pattern description
+ * @param {*} pattern pattern to extract headsign out of
+ * @returns           headsign of pattern
+ */
+export function extractHeadsignFromPattern (pattern) {
```

Comment:
There is some code in the Stop Viewer -> Schedule View that does something similar, could that be reused?

Reply:
I think if you follow the methods called by that you wind up back at this method!